castle/cms/browser/security/templates/secure-login.pt

```diff
@@ -38,7 +38,13 @@
           </div>
         </div>
       </div>
-
+      <h5 tal:define="url view/request_form_url;"
+           class="text-center"
+           style="color: green">
+        <div tal:condition="python: request.get('HTTP_REFERER').endswith(url)">
```

Comment:
what is this trying to do? Seems like putting this condition here is basically saying "if the request indicates it's come from a url that ends with the full url of the request form, then lets assume the form was submitted" which breaks down to "show a success message after a successful send" -- couldn't this come from maybe a query parameter that indicates success/failure? IE a parameter that is 'submitted' and has a value of 't' if successful? or something along those lines?